### PR TITLE
docs: add tutorial on building executor

### PR DESCRIPTION
**Pull request by cristianmtr**
Closes https://github.com/jina-ai/internal-tasks/issues/171

### CODE REVIEW

**Review by alexcg1 — APPROVED** [review]
LGTM :+1:

Still needs some rewording to make it fit with other docs, but right now I want to get it out there. I can nitpick later :)

File: docs/fundamentals/tutorials/index.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Tutorials
+
+Here you will find a selection of tutorials on Jina. 
+These guide you towards achieving a specific goal in Jina.
+
+```{toctree}
+
```
Comment (hanxiao) [inline docs/fundamentals/tutorials/index.md:7]:
```suggestion
:hidden:
```

File: docs/index.md [hunk 2]
```diff
@@ -99,7 +99,7 @@ fundamentals/concepts
 fundamentals/document/index
 fundamentals/executor/index
 fundamentals/flow/index
-fundamentals/practice-your-learning
+fundamentals/tutorials/index
```
Comment (hanxiao) [inline docs/index.md:102]: tutorials should not be part of fundamentals, as it may contain advanced topics in the future.

Please move it to a new top-level section.

**Review by hanxiao — CHANGES_REQUESTED** [review]
tutorials should not be part of fundamentals, as they may contain advanced topics in the future.

Please move it to a new top-level section.

File: docs/fundamentals/tutorials/new_executor.md [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+
+We recommend creating a [new python virtual environment](https://docs.python.org/3/tutorial/venv.html) to have a clean installation of Jina and prevent dependency clashing.
+
+We can start by installing Jina:
+
+ ```bash
+pip install jina
+ ```
+
+For more information on installing Jina, refer to this [page](https://github.com/jina-ai/jina#install).
```
Comment (alaeddine-13) [inline docs/fundamentals/tutorials/new_executor.md:21]:
````suggestion
```{admonition} See Also
:class: seealso
For more information on installing Jina, refer to the {ref}`installation page <installation>`
```
````
and in the docs installation section (get-started/install.md) do this change:
```
(installation)=
# Install
```

File: docs/fundamentals/tutorials/new_executor.md [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+
+```{admonition} Important
+:class: important
+
+It's not possible to redefine the interface of the public methods decorated by `@requests`. You can't change the name of these arguments. To see exactly which parameters you can use, check the [Executors docs](https://docs.jina.ai/fundamentals/executor/).
+```
+
+If you would like to call your `log` function only on `/index` time, you specify the endpoint with `on=`, like this:
+
+```python
```
Comment (alaeddine-13) [inline docs/fundamentals/tutorials/new_executor.md:155]:
````suggestion
```{code-block} python
---
emphasize-lines: 1
---
````

File: docs/fundamentals/tutorials/new_executor.md [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+jina hub push --public .
+```
+
+This means you will push your Executor publicly to Jina Hub. The last dot means you will use your current path. Once you run that command, you should see something like this:
+
+```{figure} ../../../.github/images/push-executor.png
+:align: center
+```
+
+Since we pushed our Executor using the `--public` flag, the only thing we will use is the ID, which, in this case, is `zsor7fe6`.
```
Comment (alaeddine-13) [inline docs/fundamentals/tutorials/new_executor.md:238]:
````suggestion
```{admonition} Note
:class: note
Since we pushed our Executor using the `--public` flag, the only thing we will use is the ID, which, in this case, is `zsor7fe6`. 
```
````
Maybe also explain what does public mean and reference the push to hub section

**Review by alaeddine-13 — COMMENTED** [review]
I also see many references done like this: `[the docs](https://docs.jina.ai/fundamentals/executor/executor-built-in-features/#handle-request-parameters)`
I personally prefer using this method : https://myst-parser.readthedocs.io/en/latest/sphinx/intro.html#reference-a-section-label-with-a-role
because it doesn't depend on the docs domain name, less prone to be broken and people who modify sections will be aware that it's referenced

File: docs/index.md (hunk 2 above)
Comment (alaeddine-13) [reply docs/index.md:102]: maybe you can merge this branch: docs-chatbot-tutorial to yours and add the executor tutorial to the same section

**Comment by cristianmtr** [comment]
@hanxiao I moved the tutorial to their own top-level section. It now looks like this

Screenshot from 2021-09-10 10-31-18

@alaeddine-13 

I refactored the links into `{ref}`s. Cool feature :) 

Please re-check

File: docs/index.md (hunk 2 above)
Comment (alaeddine-13) [reply docs/index.md]: Ignore the comment above, I'll merge yours  so that both tutorials fall in the same section

**Review by alaeddine-13 — APPROVED** [review]
LGTM

File: docs/tutorials/practice-your-learning.md [hunk 6]
```diff
@@ -1,4 +1,4 @@
-# Put into Practice
+# Putting It All Together
```
Comment (hanxiao) [inline docs/tutorials/practice-your-learning.md:1]: please revert this change and make this file stays in the original position

Comment (cristianmtr) [reply docs/tutorials/practice-your-learning.md]: Done

**Review by hanxiao — APPROVED** [review]
